openapi-integration-tests/src/test/java/io/ballerina/openapi/idl/client/IDLClientGenPluginNegativeTests.java

```diff
@@ -42,41 +47,60 @@ public class IDLClientGenPluginNegativeTests extends OpenAPITest {
     public static final Path TEST_RESOURCE = Paths.get(RESOURCES_PATH.toString() + "/client-idl-projects");
-
     @BeforeClass
     public void setupDistributions() throws IOException {
         TestUtil.cleanDistribution();
     }
 
     @Test(description = "Here uses the graphQl yaml as non openapi client")
     public void testNonOpenAPIClientDeclaration() throws IOException, InterruptedException {
-        boolean successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_07"),
+        Process process = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_07"),
                 new ArrayList<>());
         File dir = new File(RESOURCE.resolve("client-idl-projects/project_07/generated/").toString());
         Assert.assertFalse(dir.exists());
+        String msg = " ERROR [main.bal:(1:1,1:176)] no matching plugin found for client declaration";
+        compareLogOutput(process, msg);
     }
 
     @Test
     public void testInvalidSwaggerRemotePath() throws IOException, InterruptedException {
-        boolean successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_10"),
+        Process process = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_10"),
                 new ArrayList<>());
         File dir = new File(RESOURCE.resolve("client-idl-projects/project_10/generated/").toString());
         Assert.assertFalse(dir.exists());
+        String msg = "ERROR [main.bal:(2:5,3:13)] unable to get resource from uri, reason: ";
+        compareLogOutput(process, msg);
     }
 
     @Test
     public void testInvalidSwaggerLocalPath() throws IOException, InterruptedException {
-        boolean successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_11"),
+        Process process = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_11"),
                 new ArrayList<>());
         File dir = new File(RESOURCE.resolve("client-idl-projects/project_11/generated/").toString());
         Assert.assertFalse(dir.exists());
+        String msg = "ERROR [main.bal:(2:5,2:35)] unable to get resource from uri, reason: no protocol: openapi2.yaml";
+        compareLogOutput(process, msg);
     }
 
     @Test
     public void testInvalidSwaggerContract() throws IOException, InterruptedException {
-        boolean successful = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_12"),
+        Process process = executeRun(DISTRIBUTION_FILE_NAME, TEST_RESOURCE.resolve("project_12"),
                 new ArrayList<>());
         File dir = new File(RESOURCE.resolve("client-idl-projects/project_12/generated/").toString());
         Assert.assertFalse(dir.exists());
+        String msg = " ERROR [main.bal:(2:5,2:34)] OpenAPI definition has errors:";
+        compareLogOutput(process, msg);
+    }
+
+    private static void compareLogOutput(Process process, String msg) throws IOException {
```

Comment:
Shall we have a name like below as we are only checking the error stream?
```suggestion
    private static void assertOnErrorStream(Process process, String msg) throws IOException {
```

Reply:
+1